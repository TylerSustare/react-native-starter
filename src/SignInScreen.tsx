import React from 'react';
import { View } from 'react-native';
import { Button, TextInput } from 'react-native-paper';
import { withFirebaseHOC, IFirebase } from './firebase';

interface Props {
  firebase: IFirebase;
}

export const SignInScreen: React.FC<Props> = props => {
  const { firebase } = props;
  const [username, setUsername] = React.useState('');
  const [password, setPassword] = React.useState('');

  return (
    <View>
      <TextInput placeholder="Email Address" testID="username-input" value={username} onChangeText={setUsername} />
      <TextInput placeholder="Password" testID="password-input" value={password} onChangeText={setPassword} secureTextEntry />
      <Button testID="signup-button" onPress={() => firebase.signupWithEmail(username, password)}>
        Sign Up
      </Button>
      <Button testID="login-button" onPress={() => firebase.loginWithEmail(username, password)}>
        Sign In
      </Button>
    </View>
  );
};

export default withFirebaseHOC(SignInScreen);
