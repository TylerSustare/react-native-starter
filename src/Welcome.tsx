import React, { useEffect, useState } from 'react';
import { Text, View } from 'react-native';
import { withFirebaseHOC, IFirebase } from './firebase';
import { Button } from 'react-native-paper';

interface Props {
  firebase: IFirebase;
}

const Welcome: React.FC<Props> = ({ firebase }) => {
  const [user, setUser] = useState(null);
  useEffect(() => {
    let unsubscribe;
    const getUser = async () => {
      unsubscribe = await firebase.checkUserAuth(user => setUser(user));
    };
    getUser();
    return function cleanup() {
      unsubscribe();
    };
  }, []);
  return (
    <View>
      <Text>Welcome {user?.email}</Text>
      <Button onPress={() => firebase.signOut()}> signout </Button>
    </View>
  );
};

export default withFirebaseHOC(Welcome);
