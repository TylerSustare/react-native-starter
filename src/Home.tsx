import React, { useState, useEffect } from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import SignInScreen from './SignInScreen';
import Welcome from './Welcome';
import { NavigationContainer } from '@react-navigation/native';
import { withFirebaseHOC, IFirebase } from './firebase';

const Stack = createStackNavigator();

interface Props {
  firebase: IFirebase;
}

const App: React.FC<Props> = ({ firebase }) => {
  const [user, setUser] = useState(null);
  useEffect(() => {
    let unsubscribe;
    const getUser = async () => {
      unsubscribe = await firebase.checkUserAuth(user => setUser(user));
    };
    getUser();
    return function cleanup() {
      unsubscribe();
    };
  }, []);
  return (
    <NavigationContainer>
      <Stack.Navigator>
        {user == null ? (
          <Stack.Screen
            name="SignIn"
            component={SignInScreen}
            options={{
              title: 'Sign in',
              headerStyle: {
                backgroundColor: '#59bfff'
              },
              headerTintColor: '#fff',
              headerTitleStyle: {
                fontWeight: 'bold'
              }
            }}
          />
        ) : (
          <Stack.Screen name="Welcome" component={Welcome} />
        )}
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default withFirebaseHOC(App);
