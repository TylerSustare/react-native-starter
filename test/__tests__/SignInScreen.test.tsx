import React from 'react';
import { fireEvent, render, wait } from '@testing-library/react-native';
import { SignInScreen } from '../../src/SignInScreen';
import { FuegoBase } from '../mocks/firebase_mocks';
import sinon from 'sinon';

let sandbox: sinon.SinonSandbox;
beforeEach(() => {
  sandbox = sinon.createSandbox();
});
afterEach(() => {
  sandbox.restore();
});

const testEmail = 'spoon@spoon.spoon';
const testPassword = 'spoon 1';
test('SignIn screen has a form that calls sign in', async () => {
  const f = new FuegoBase();
  // @ts-ignore
  sandbox.spy(f);
  const { getByTestId, getByText, queryByTestId } = render(<SignInScreen firebase={f} />);

  const input = getByTestId('username-input');
  fireEvent.changeText(input, testEmail);

  await wait(() => expect(queryByTestId('username-input')).toBeTruthy());

  const passInput = getByTestId('password-input');
  fireEvent.changeText(passInput, testPassword);

  const result = getByTestId('username-input');
  expect(result).toMatchSnapshot();
  expect(result.props.value).toBe(testEmail);

  const signup = getByText('Sign Up');
  fireEvent.press(signup);

  // @ts-ignore
  expect(f.signupWithEmail.calledOnce).toEqual(true);
});

test('SignIn screen has a form that does stuff', async () => {
  const f = new FuegoBase();
  // @ts-ignore
  sandbox.spy(f);
  const { getByTestId, getByText, queryByTestId } = render(<SignInScreen firebase={f} />);
  const testEmail = 'spoon@spoon.spoon';
  const testPassword = 'spoon 1';

  const input = getByTestId('username-input');
  fireEvent.changeText(input, testEmail);

  await wait(() => expect(queryByTestId('username-input')).toBeTruthy());

  const passInput = getByTestId('password-input');
  fireEvent.changeText(passInput, testPassword);

  const result = getByTestId('username-input');
  expect(result).toMatchSnapshot();
  expect(result.props.value).toBe(testEmail);

  const signup = getByText('Sign Up');
  fireEvent.press(signup);

  // @ts-ignore
  expect(f.signupWithEmail.calledOnce).toEqual(true);
});
