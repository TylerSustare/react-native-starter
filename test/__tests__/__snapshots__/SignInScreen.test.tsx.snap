// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn screen has a form that calls sign in 1`] = `
<TextInput
  allowFontScaling={true}
  editable={true}
  multiline={false}
  placeholder="Email Address"
  placeholderTextColor="rgba(0, 0, 0, 0.54)"
  rejectResponderTermination={true}
  selectionColor="#6200ee"
  style={
    Array [
      Object {
        "flexGrow": 1,
        "margin": 0,
        "textAlign": "left",
        "zIndex": 1,
      },
      Object {
        "paddingHorizontal": 12,
      },
      Object {
        "height": 64,
      },
      Object {
        "paddingBottom": 0,
        "paddingTop": 0,
      },
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": undefined,
        "textAlignVertical": "center",
      },
    ]
  }
  testID="username-input"
  underlineColorAndroid="transparent"
  value="spoon@spoon.spoon"
/>
`;

exports[`SignIn screen has a form that does stuff 1`] = `
<TextInput
  allowFontScaling={true}
  editable={true}
  multiline={false}
  placeholder="Email Address"
  placeholderTextColor="rgba(0, 0, 0, 0.54)"
  rejectResponderTermination={true}
  selectionColor="#6200ee"
  style={
    Array [
      Object {
        "flexGrow": 1,
        "margin": 0,
        "textAlign": "left",
        "zIndex": 1,
      },
      Object {
        "paddingHorizontal": 12,
      },
      Object {
        "height": 64,
      },
      Object {
        "paddingBottom": 0,
        "paddingTop": 0,
      },
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": undefined,
        "textAlignVertical": "center",
      },
    ]
  }
  testID="username-input"
  underlineColorAndroid="transparent"
  value="spoon@spoon.spoon"
/>
`;
